### PR TITLE
Adds a test for checking vtorc fixes semi-sync correctly

### PR DESCRIPTION
**Pull request by GuptaManan100**
## Description

This PR adds a test to the vtorc test suite verifying that vtorc fixes semi-sync on a tablet if it is not correctly set.

## Related Issue(s)

- #6769 

## Checklist
- [ ] Should this PR be backported?
- [x] Tests were added or are not required
- [x] Documentation was added or is not required

### CODE REVIEW

File: go/test/endtoend/vtorc/general/vtorc_test.go [hunk 1]
```diff
@@ -286,3 +288,68 @@ func TestCircularReplication(t *testing.T) {
 	// check that the writes still succeed
 	utils.VerifyWritesSucceed(t, clusterInfo, primary, []*cluster.Vttablet{replica}, 10*time.Second)
 }
+
+// TestSemiSync tests that semi-sync is setup correctly by vtorc if it is incorrectly set
+func TestSemiSync(t *testing.T) {
+	// stop any vtorc instance running due to a previous test.
+	utils.StopVtorc(t, clusterInfo)
```
Comment (deepthi) [inline go/test/endtoend/vtorc/general/vtorc_test.go:295]: I'm surprised you need this. Should a previous `vtorc` not be torn down at the end of the previous test?

File: go/test/endtoend/vtorc/general/vtorc_test.go [hunk 2]
```diff
@@ -286,3 +288,68 @@ func TestCircularReplication(t *testing.T) {
...
+	_, err = utils.RunSQL(t, "SET GLOBAL rpl_semi_sync_slave_enabled = 1", rdonly, "")
+	require.NoError(t, err)
+	_, err = utils.RunSQL(t, "SET GLOBAL rpl_semi_sync_master_enabled = 0", primary, "")
+	require.NoError(t, err)
+
+	timeout := time.After(20 * time.Second)
+	for {
+		select {
+		case <-timeout:
+			t.Fatal("timedout waiting for semi sync settings to be fixed")
```
Comment (deepthi) [inline go/test/endtoend/vtorc/general/vtorc_test.go:343]: nit: "timed out".
we should use `require.Fail` and not use any of the `t.` funcs at all.

File: go/test/endtoend/vtorc/utils/utils.go [hunk 3]
```diff
@@ -744,3 +744,93 @@ func MakeAPICallUntilRegistered(t *testing.T, url string) (status int, response
...
+				t.Fatalf("Error starting start mysql: %v", err)
+			}
+			mysqlCtlProcessList = append(mysqlCtlProcessList, proc)
+		}
+	}
+
+	// Wait for mysql processes to start
+	for _, proc := range mysqlCtlProcessList {
+		if err := proc.Wait(); err != nil {
+			t.Fatalf("Error starting mysql: %v", err)
```
Comment (deepthi) [inline go/test/endtoend/vtorc/utils/utils.go:799]: `require.NoError`

File: go/test/endtoend/vtorc/utils/utils.go [hunk 4]
```diff
@@ -744,3 +744,93 @@ func MakeAPICallUntilRegistered(t *testing.T, url string) (status int, response
...
+
+	//Start MySql
+	var mysqlCtlProcessList []*exec.Cmd
+	for _, shard := range clusterInstance.Keyspaces[0].Shards {
+		for _, tablet := range shard.Vttablets {
+			log.Infof("Starting MySql for tablet %v", tablet.Alias)
+			proc, err := tablet.MysqlctlProcess.StartProcess()
+			if err != nil {
+				t.Fatalf("Error starting start mysql: %v", err)
+			}
```
Comment (deepthi) [inline go/test/endtoend/vtorc/utils/utils.go:791]: `require.NoError`

File: go/test/endtoend/vtorc/utils/utils.go [hunk 5]
```diff
@@ -241,18 +241,25 @@ func demotePrimaryTablet(ts *topo.Server) (err error) {
...
-// startVtorc is used to start the orchestrator with the given extra arguments
-func startVtorc(t *testing.T, clusterInfo *VtOrcClusterInfo, orcExtraArgs []string, pathToConfig string) {
+// StartVtorc is used to start the orchestrator with the given extra arguments
+func StartVtorc(t *testing.T, clusterInfo *VtOrcClusterInfo, orcExtraArgs []string, configFileName string) {
 	t.Helper()
+	workingDir := os.Getenv("PWD")
+	idx := strings.Index(workingDir, "vtorc")
+	if idx == -1 {
+		t.Fatalf("SetupVttabletsAndVtorc should only be used from a package inside the vtorc directory")
+	}
```
Comment (deepthi) [inline go/test/endtoend/vtorc/utils/utils.go:251]: use `require`.

File: go/test/endtoend/vtorc/general/vtorc_test.go (hunk 1 above)
Comment (GuptaManan100) [reply go/test/endtoend/vtorc/general/vtorc_test.go:295]: The previous tests actually don't cleanup their `vtorc` instances. All the tests are responsible for removing any previous instance of vtorc, vttablets and cleaning MySQL. This test is using a new clusterInstance so it has to remove any previously running vtorc

File: go/test/endtoend/vtorc/general/vtorc_test.go (hunk 2 above)
Comment (GuptaManan100) [reply go/test/endtoend/vtorc/general/vtorc_test.go]: Fixed

File: go/test/endtoend/vtorc/utils/utils.go (hunk 3 above)
Comment (GuptaManan100) [reply go/test/endtoend/vtorc/utils/utils.go]: Fixed

File: go/test/endtoend/vtorc/utils/utils.go (hunk 4 above)
Comment (GuptaManan100) [reply go/test/endtoend/vtorc/utils/utils.go]: Fixed

File: go/test/endtoend/vtorc/utils/utils.go (hunk 5 above)
Comment (GuptaManan100) [reply go/test/endtoend/vtorc/utils/utils.go]: Fixed

**Pull request merged**